tests/test_permissions.py

```diff
@@ -0,0 +1,151 @@
+
+    @staticmethod
+    @responses.activate
+    def test_create_permission_success(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=404,
+        )
+        responses.add(
+            responses.PUT,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=201,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        with pytest.raises(PermissionNotFoundException):
+            artifactory_permission.create(PERMISSION)
+
+            artifactory_permission.get.assert_called_with(PERMISSION.name)
+        assert artifactory_permission.get.call_count == 2
+
+    @staticmethod
+    @responses.activate
+    def test_get_permission_error_not_found():
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            status=404,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        with pytest.raises(PermissionNotFoundException):
+            artifactory_permission.get(PERMISSION.name)
+
+    @staticmethod
+    @responses.activate
+    def test_get_permission_success(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=200,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        artifactory_permission.get(PERMISSION.name)
+
+        artifactory_permission.get.assert_called_with(PERMISSION.name)
+
+    @staticmethod
+    @responses.activate
+    def test_list_group_success(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions",
+            json=[SIMPLE_PERMISSION.dict()],
+            status=200,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "list")
+        artifactory_permission.list()
+
+        artifactory_permission.list.assert_called_once()
+
+    @staticmethod
+    @responses.activate
+    def test_delete_permission_fail_if_group_not_found(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            status=404,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+
+        with pytest.raises(PermissionNotFoundException):
+            artifactory_permission.delete(PERMISSION.name)
+
+            artifactory_permission.get.assert_called_once_with(PERMISSION.name)

```

Comment:
I *think* this assert should be pulled out of the `with` block, because if the previous call raises I think it stops the execution and doesn't run the subsequent assert.